gputop/remote/gputop.js

```diff
@@ -773,44 +772,69 @@ Gputop.prototype.process_features = function(features){
-    $.get(gputop.xml_file_name_, function (xml) {
-        gputop.parse_xml_metrics(xml);
-
-        if (gputop_is_demo())
-            gputop.metrics_.forEach(function (metric) { metric.supported_ = true; });
-        else {
-            gputop.metrics_.forEach(function (metric) { metric.supported_ = false; });
-
-            if (features.supported_oa_query_guids.length == 0) {
-                gputop_ui.show_alert("No OA metrics are supported on this Kernel " +
-                                     features.get_kernel_release(), "alert-danger");
-            } else {
-                features.supported_oa_query_guids.forEach(function (guid, i, a) {
-                    var metric = gputop.get_map_metric(guid);
-                    metric.supported_ = true;
-                    metric.print();
-                });
+
+    if (!is_nodejs) {
+        $.get(gputop.xml_file_name_, function (xml) {
+            gputop.parse_xml_metrics(xml);
+
+            if (gputop_is_demo())
+                gputop.metrics_.forEach(function (metric) { metric.supported_ = true; });
+            else {
+                gputop.metrics_.forEach(function (metric) { metric.supported_ = false; });
+
+                if (features.supported_oa_query_guids.length == 0) {
+                    gputop_ui.show_alert("No OA metrics are supported on this Kernel " +
+                                        features.get_kernel_release(), "alert-danger");
+                } else {
+                    features.supported_oa_query_guids.forEach(function (guid, i, a) {
+                        var metric = gputop.get_map_metric(guid);
+                        metric.supported_ = true;
+                        metric.print();
+                    });
+                }
             }
-        }
 
-        gputop_ui.update_features(features);
-    });
+            gputop_ui.update_features(features);
+        });
+    } else {
+        http_get(gputop.xml_file_name_, function (xml) {
+            gputop.parse_xml_metrics(xml);
+
+            if (gputop_is_demo())
+                gputop.metrics_.forEach(function (metric) { metric.supported_ = true; });
+            else {
+                gputop.metrics_.forEach(function (metric) { metric.supported_ = false; });
+
+                if (features.supported_oa_query_guids.length == 0) {
+                    gputop_ui.show_alert("No OA metrics are supported on this Kernel " +
+                                        features.get_kernel_release(), "alert-danger");
+                } else {
+                    features.supported_oa_query_guids.forEach(function (guid, i, a) {
+                        var metric = gputop.get_map_metric(guid);
+                        metric.supported_ = true;
+                        metric.print();
+                    });
+                }
+            }
+
+            gputop_ui.update_features(features);
+        }, function (error) { console.log(error); });
+    }
```

Comment:
It looks like the only difference between these two blocks is using http_get() for nodejs, but I guess we should be able to use http_get in both cases to avoid a special case here.